types/index.d.ts

```diff
@@ -0,0 +1,101 @@
+// TypeScript Version: 2.4
+import Vue, { PluginFunction } from 'vue'
+
+/**
+ * Covers beforeCreate(), beforeDestroy() and data().
+ *
+ * No public members.
+ */
+export const validationMixin: any
+
+// const Validation
+export interface Validation extends Vue {
+    // const validationGetters
+    readonly $invalid: boolean
+    readonly $dirty: boolean
+    readonly $error: boolean
+    readonly $pending: boolean
+    readonly $params: { [attr: string]: any }
+
+    // const validationMethods
+    $touch(): never
+    $reset(): never
+    $flattenparams(): Params[]
```

Comment:
This seems to be a typo. Should be `$flattenParams()`.

See here: https://github.com/monterail/vuelidate/blob/a6e74f4d2b8c16a3396b1fe141729412fcd5cf0d/src/index.js#L135

Also looks like `Params` is not the correct type for this. I am getting back an object that contains the `params: Params`, along with `name` and `path`.

Screenshot: 
![screenshot_256](https://user-images.githubusercontent.com/6885927/40316838-95f7f0e8-5ced-11e8-971f-3d84970f5e78.png)

Reply:
adjusted types accordingly.

Reply:
Thank you @cypherix93.
I fixed the mistake in the PR.

About "Params" i share your doubt. It is coupled to withParams(..) and i have seen examples where there is more than the "type" attribute. Besides it has changed lately.

Seems like the whole 
"import { helpers } from 'vuelidate/lib/validators'"
is new and actually uncovered.